### DISCUSSION TITLE
Quais são bons exemplos de código es6/es2015 para se inspirar?

### DISCUSSION THREAD

**Discussion by diegoemerique** [frontendbr/forum#396] [Discussão]
Olá Galera,

Estou muito em dúvida de como as coisas ficaram a partir do es6, pois antes existiam muitas técnicas diferentes e não to conseguindo mensurar se estou indo em um caminho certo ou não.

Procurando no Github eu só acho projeto muito gigante com milhares de pessoas mexendo e que resolvem problemas bizarros, ou coisas muito especificas. Não consegui achar um projeto de um site "normal", ou de uma UI. E Quando acha é um React ou angular da vida.

Exemplo de coisas q eu fazia antes e que estou em dúvida agr.

Module Pattern:

```

var Modulo = (function() {
   var whatever = 'whatever';

   publicApi = {
      start: function() {
         // ...
      },
      eventHandler: function() {
        // ...  
      },
     getFoo: function() {
       return 'foo';
     }
  }

  return publicApi;
})();

Modulo.start();

Modulo.getFoo();

```

Algo +/- assim.. Como fica isso de um modo mais robusto e legal com o es6?

outro tipo de coisa q eu via usar era:

```

(function( window, undefined ) {

  var Coisa = function( whatever ) {
    return new Coisa.init( whatever );
  };
  
  Coisa.prototype = {
    getFoo: function() {
       return 'foo';
     }
    // ..
  }

  Coisa.init = function( whatever ) {
    // ...
    return this;
  };

  Coisa.init.prototype = Coisa.prototype;

  window.Coisa= Coisa;

  return Coisa;

})( window );

```

O negocio é, onde achar bons exemplos, da atualidade, onde podemos entrar ler e conquistar técnicas novas sendo usadas na vida real.

**Comment by victormiguez** [comment]
Legal a ideia do tópico, gostaria de ter alguns exemplos também!
Um projeto que eu gosto muito do código é o [cep-promise](https://github.com/filipedeschamps/cep-promise).

**Comment by diegoemerique** [comment]
@victormiguez isso mesmo!

Quem poder e tiver exemplos de bons códigos em es6, como esse do CEP, cola ai que vai ajudar muito o pessoal q ta começando a querer organizar melhor seu código com es6!

**Comment by mksanti** [comment]
https://github.com/tastejs/todomvc/tree/gh-pages/examples/vanilla-es6

**Comment by fdaciuk** [comment]
Se quiser um bom suporte para browsers, o ideal é sempre usar um transpiler (babel, por exemplo).

Mas sobre a ideia de como utilizar módulos, pode continuar usando como você já fazia, ou usar ferramentas para transformar seu código usando ES6 modules para AMD, UMD, ou só adicionar uma IIFE.

Com webpack é "tranquilo" de fazer isso. Tranquilo no sentido de que ele resolve qualquer problema para qualquer dependência de browser (JS, CSS e HTML), mas sempre vai ter algumas configurações pra fazer, dependendo do tipo do teu projeto.

**Comment by rafaeljesus** [comment]
De uma olhada em projetos como [lodash](https://github.com/lodash/lodash), [ramda](https://github.com/ramda/ramda) e eu particularmente adoro e escrevo code nesse style [redux-saga](https://github.com/redux-saga/redux-saga/blob/master/src/internal/channel.js).

Estude um pouco de programação funcional o @suissa tem material free e leia tbm os excelentes posts do @ericelliott no medium
